zetaclient/chains/solana/signer/signer.go

```diff
@@ -299,8 +315,21 @@ func (signer *Signer) broadcastOutbound(
 				logger.Error().Err(err).Fields(lf).Msgf("unable to deserialize PDA info")
 			} else if pda.Nonce > nonce {
 				logger.Info().Err(err).Fields(lf).Msgf("PDA nonce %d is greater than outbound nonce, stop retrying", pda.Nonce)
+				if nonce > signer.maxProcessedNonce.Load() {
+					signer.maxProcessedNonce.Store(nonce)
+				}
 				break
 			}
+			if signer.maxProcessedNonce.Load() == 0 {
+				signer.maxProcessedNonce.Store(pda.Nonce)
```

Comment:
`pda` is a invalid value when program execution comes here. Ideally, it would be good if we have `maxProcessedNonce` updating logic in one single place (including initialization and update), like

```
if nonce >= signer.maxProcessedNonce.Load() {
	signer.maxProcessedNonce.Store(nonce)
}
```